core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java

```diff
@@ -263,6 +264,17 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
 
             @Override
             public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+                if (handshakeFailed == null) {
+                    // An exception was raised before the handshake event was completed.
+                    // A legacy HTTPS server such as Microsoft-IIS/8.5 may reset the connection
+                    // if no cipher suites in common.
+                    final IllegalStateException maybeHandshakeException = new IllegalStateException(
+                            "An unexpected exception during TLS handshake. One possible reason behind the " +
+                            "failure is no cipher suites in common. " +
+                            "ciphers: " + sslCtx.cipherSuites(), cause);
```

Comment:
How about this: (Not a complete diff :sweat_smile:)
```suggestion
                    final IllegalStateException maybeHandshakeException = new IllegalStateException(
                            "An unexpected exception during TLS handshake. Possible reasons: no cipher suites in common, unsupported TLS version, etc. " +
                            "(TLS version: " ... ", cipher suites: " + sslCtx.cipherSuites() + ')', cause);
```

Reply:
Looks much better!  🙇‍♂️